common/src/main/java/com/wynntils/features/EmeraldPouchHotkeyFeature.java

```diff
@@ -0,0 +1,197 @@
+                            slotStack = pouchStack;
+                        }
+                        if (slotNumber < 9) {
+                            // sendPacket uses raw slot numbers, we need to remap the hotbar
+                            slotNumber += 36;
+                        }
+                        changedSlots.putIfAbsent(slotNumber, slotStack);
+                        McUtils.player()
+                                .connection
+                                .send(new ServerboundContainerClickPacket(
+                                        player.inventoryMenu.containerId,
+                                        player.inventoryMenu.getStateId(),
+                                        slotNumber,
+                                        1,
+                                        ClickType.PICKUP,
+                                        ItemStack.EMPTY,
+                                        changedSlots));
+                    }
+                    case 0 -> McUtils.sendMessageToClient(
+                            new TranslatableComponent("feature.wynntils.emeraldPouchKeybind.noPouch")
+                                    .withStyle(ChatFormatting.DARK_RED));
+
+                    case 1 -> {
+                        int slotNumber = emeraldPouches
+                                .entrySet()
+                                .iterator()
+                                .next()
+                                .getKey(); // Just get the first value in the HashMap since we only have one value
+
+                        if (slotNumber < 9) {
+                            // sendPacket uses raw slot numbers, we need to remap the hotbar
+                            slotNumber += 36;
+                        }
+                        changedSlots.putIfAbsent(
+                                slotNumber, player.getInventory().getItem(slotNumber));
+                        McUtils.player()
+                                .connection
+                                .send(new ServerboundContainerClickPacket(
+                                        player.inventoryMenu.containerId,
+                                        player.inventoryMenu.getStateId(),
+                                        slotNumber,
+                                        1,
+                                        ClickType.PICKUP,
+                                        ItemStack.EMPTY,
+                                        changedSlots));
+                    }
+                }
+            });
+
+    private int getPouchUsage(ItemStack stack) { // TODO: move to EmeraldPouchManager or equivalent when created
+        String lore = ItemUtils.getStringLore(stack);
+        Matcher usageMatcher = POUCH_USAGE_PATTERN.matcher(lore);
+        if (!usageMatcher.find()) {
+            if (lore.contains("§7Empty")) {
+                return 0;
+            }
+
+            return -1;
+        }
```

Comment:
It might be nice to log something, or even disable the feature as a failsafe if that ever gets done